aws/resource_aws_datasync_location_s3.go

```diff
@@ -51,6 +51,12 @@ func resourceAwsDataSyncLocationS3() *schema.Resource {
 					},
 				},
 			},
+			"s3_storage_class": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
Should be `Optional: true`?
According to the [AWS API Reference](https://docs.aws.amazon.com/datasync/latest/userguide/API_CreateLocationS3.html#API_CreateLocationS3_RequestSyntax) this argument is not required.

Reply:
Since the default value for S3-outpost and s3 is different, thought of keeping is required, otherwise it might be a breaking change in future.

Reply:
Given that there will be existing resources deployed with differing defaults, the way to handle this is with a `Computed/Optional` attribute:

```go
			"s3_storage_class": {
				Type:         schema.TypeString,
				Optional:     true,
				Computed:     true,
				ForceNew:     true,
				ValidateFunc: validation.StringInSlice(datasync.S3StorageClass_Values(), false),
			},
```

Reply:
Agreed, updated the code.